File: Tugas 1/Polymorphism.php
<?php

// Interface
interface Binatang {
    public function bersuara();
}

// Kelas kucing yang mengimplementasikan interface Binatang
class Kucing implements Binatang {
    public function bersuara() {
        return "Meong";
    }
}

// Kelas anjing yang juga mengimplementasikan interface Binatang
class Anjing implements Binatang {
    public function bersuara() {
        return "Guk Guk";
    }
}

// Fungsi yang menerima objek Binatang sebagai parameter
function suaraHewan(Binatang $hewan) {
    echo $hewan->bersuara() . "<br>";
}

// Contoh penggunaan
$kucing = new Kucing();
$anjing = new Anjing();

suaraHewan($kucing); // Output: Meong
suaraHewan($anjing); // Output: Guk Guk

?>

File: Tugas 1/Polymorphism.py
class Binatang:
    def suara(self):
        pass

class Kucing(Binatang):
    def suara(self):
        return "Meow!"

class Anjing(Binatang):
    def suara(self):
        return "Guk Guk"


# Fungsi yang menggunakan polimorfisme
def dengarkan_suara(binatang):
    print(binatang.suara())

# Membuat instance dari kelas-kelas yang berbeda
kucing = Kucing()
anjing = Anjing()

# Memanggil fungsi dengan berbagai instance
dengarkan_suara(kucing)  # Output: Meow!
dengarkan_suara(anjing)  # Output: Woof!